rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -65,12 +65,19 @@ get_node_names(
     RMW_SET_ERROR_MSG("failed to get default participant qos");
     return RMW_RET_ERROR;
   }
-  node_names->data[0] = rcutils_strdup(participant_qos.participant_name.name, allocator);
-  if (!node_names->data[0]) {
+
+  rcutils_string_array_t tmp_names_list = rcutils_get_zero_initialized_string_array();
+  rcutils_ret = rcutils_string_array_init(&tmp_names_list, length, &allocator);
+
+  tmp_names_list.data[0] = rcutils_strdup(participant_qos.participant_name.name, allocator);
+
+  if (!tmp_names_list.data[0]) {
     RMW_SET_ERROR_MSG("could not allocate memory for node name")
     return RMW_RET_BAD_ALLOC;
   }
 
+  int named_nodes_num = 0;
```

Comment:
doesn't this need to start at 1 since we've already added the 0th entry?

Reply:
fixed. yes, I think that's better